modules/columns/main.tf

```diff
@@ -1,6 +1,53 @@
-resource "honeycombio_column" "columns" {
-  count   = length(var.columns)
-  name    = keys(var.columns)[count.index]
-  type    = values(var.columns)[count.index]
-  dataset = var.dataset_name
+resource "null_resource" "create_metrics_columns" {
+  provisioner "local-exec" {
+    command = <<-EOT
+      cat > metrics-event.json <<EOF
+{
+%{for m in var.metrics_columns~}
+%{if m != element(var.metrics_columns, length(var.metrics_columns) - 1)~}
+%{if m != "hostname"~}
+"${m}": 0.0,
+%{else~}
+"${m}": "",
+%{endif~}
+%{else~}
+%{if m != "hostname"~}
+"${m}": 0.0
+%{else~}
+"${m}": ""
+%{endif~}
+%{endif~}
+%{endfor~}
+}
+EOF
+
+      HONEYCOMB_DATASET="${var.metrics_dataset}"
+
+      curl https://api.honeycomb.io/1/events/$HONEYCOMB_DATASET -X POST \
```

Comment:
So one thing about this -  it's very likely that subsequent steps that depend on the new columns will fail, because it usually takes 1-10sec for new columns to appear queryable. 

You may need a wait loop curl + sleep here

Reply:
Submitted a fix